platform/linuxbsd/display_server_x11.cpp

```diff
@@ -1850,23 +1850,38 @@ DisplayServerX11::LatinKeyboardVariant DisplayServerX11::get_latin_keyboard_vari
 	_THREAD_SAFE_METHOD_
 
 	XkbDescRec *xkbdesc = XkbAllocKeyboard();
+	if (!xkbdesc)
+		WARN_PRINT("XkbAllocKeyboard returned null. Defaulting to QWERTY.");
 	ERR_FAIL_COND_V(!xkbdesc, LATIN_KEYBOARD_QWERTY);
 
 	XkbGetNames(x11_display, XkbSymbolsNameMask, xkbdesc);
+	if (!xkbdesc->names)
+		WARN_PRINT("xkbdesc->names null. Defaulting to QWERTY");
 	ERR_FAIL_COND_V(!xkbdesc->names, LATIN_KEYBOARD_QWERTY);
+	if (!xkbdesc->names->symbols)
+		WARN_PRINT("xkbdesc->names->symbols null. Defaulting to QWERTY.");
 	ERR_FAIL_COND_V(!xkbdesc->names->symbols, LATIN_KEYBOARD_QWERTY);
 
+	// To check the xkb_symbols from a bash: some examples:
+	// $ setxkbmap -print | grep xkb_symbols
+	// xkb_symbols   { include "pc+fr+us:2+inet(evdev)"        };
+	// xkb_symbols   { include "pc+us+ru:2+inet(evdev)+capslock(grouplock)+terminate(ctrl_alt_bksp)"   };
+	// Refs: https://medium.com/@damko/a-simple-humble-but-comprehensive-guide-to-xkb-for-linux-6f1ad5e13450
 	char *layout = XGetAtomName(x11_display, xkbdesc->names->symbols);
+	if (!layout)
+		WARN_PRINT("XGetAtomName returned null. Defaulting to QWERTY.");
 	ERR_FAIL_COND_V(!layout, LATIN_KEYBOARD_QWERTY);
 
 	Vector<String> info = String(layout).split("+");
+	if (info.size() < 2)
+		WARN_PRINT(String("XGetAtomName layout too short: '") + String(layout) + String("' : Defaulting to QWERTY."));
 	ERR_FAIL_INDEX_V(1, info.size(), LATIN_KEYBOARD_QWERTY);
 
 	if (info[1].find("colemak") != -1) {
 		return LATIN_KEYBOARD_COLEMAK;
 	} else if (info[1].find("qwertz") != -1) {
 		return LATIN_KEYBOARD_QWERTZ;
-	} else if (info[1].find("azerty") != -1) {
+	} else if (info[1].find("azerty") != -1 || info[1] == "fr") {
```

Comment:
Is "fr" check correct for all variants of French keyboard layout? At least on macOS there are Canadian French and Swiss French layouts that are QWERTY and QWERTZ with extra diacritics.

Reply:
It seems to be, at least on KDE/X11 it seems the variant called only `fr` would be "French (Default)", which I assume is similar to `fr(azerty)`.

Here's a test:
![Screenshot_20200610_114452](https://user-images.githubusercontent.com/4701338/84253104-cec38280-ab0f-11ea-9d39-2cdfbb6a27eb.png)
```
$ setxkbmap -print | grep xkb_symbols
        xkb_symbols   { include "pc+fr(us)+fr:2+fr(azerty):3+fr(oss):4+inet(evdev)+terminate(ctrl_alt_bksp)+group(lalt_lshift_toggle)+compose(ralt)"    };
```
Only shows first 4 "main" layouts, can't increase that number, so here's for the others tested:
![Screenshot_20200610_114554](https://user-images.githubusercontent.com/4701338/84253212-f581b900-ab0f-11ea-9efb-92ac8d4b7ead.png)
```
$ setxkbmap -print | grep xkb_symbols
        xkb_symbols   { include "pc+fr(us)+fr(oss):2+ca:3+fr(afnor):4+inet(evdev)+terminate(ctrl_alt_bksp)+group(lalt_lshift_toggle)+compose(ralt)"     };
```

But this also shows that for e.g. German there's virtually no layout that properly reports itself as `qwertz` (apart from `neo_qwertz`... which should maybe be `LATIN_KEYBOARD_NEO`?):
![Screenshot_20200610_114854](https://user-images.githubusercontent.com/4701338/84253539-5e693100-ab10-11ea-940b-c8cc3e8f2a5d.png)
```
$ setxkbmap -print | grep xkb_symbols
        xkb_symbols   { include "pc+us+de:2+de(deadacute):3+de(neo_qwertz):4+inet(evdev)+terminate(ctrl_alt_bksp)+group(lalt_lshift_toggle)+compose(ralt)"      };
```

All in all this logic seems very brittle and I'd be in favor of outright dropping it instead of trying to fix it. If we want accurate keyboard layout detection, we should likely rely on a cross-platform library that can do this for us.